eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,506 @@
+                        parameters:
+                          FolderForUpload: '$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}'
+                          BlobSASKey: '$(azure-sdk-docs-prod-sas)'
+                          BlobName: '$(azure-sdk-docs-prod-blob-name)'
+                          TargetLanguage: 'java'
+                          ArtifactLocation: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
+                          # we override the regular script path because we have cloned the build tools repo as a separate artifact.
+                          ScriptPath: 'eng/common/scripts/copy-docs-to-blobstorage.ps1'
+
+  - ${{if ne(parameters.EnableIntegrationStage, false)}}:
+    - stage: Integration
+      dependsOn: Signing
+      jobs:
+      - job: PublishPackages
+        condition: or(eq(variables['SetDevVersion'], 'true'), and(eq(variables['Build.Reason'],'Schedule'), eq(variables['System.TeamProject'], 'internal')))
+        displayName: Publish package to daily feed
+        variables:
+          - template: /eng/pipelines/templates/variables/globals.yml
+          - name: RepositoryUrl
+            value: https://pkgs.dev.azure.com/azure-sdk/internal/_packaging/azure-sdk-for-java-pr/maven/v1
+
+        pool:
+          name: azsdk-pool-mms-win-2019-general
+          vmImage: windows-2019
+        steps:
+        - checkout: azure-sdk-build-tools
+          path: azure-sdk-build-tools
+        - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+          parameters:
+            SkipDefaultCheckout: true
+            Repositories:
+              - Name: Azure/azure-sdk-for-java
+                Commitish: $(Build.SourceVersion)
+                WorkingDirectory: $(Pipeline.Workspace)/azure-sdk-for-java
+
+        - download: current
+          displayName: 'Download Artifact: ${{parameters.ArtifactName}}-signed'
+          artifact: ${{parameters.ArtifactName}}-signed
+        - pwsh: |
+            # For safety default to publishing to the private feed.
+            # Publish to https://pkgs.dev.azure.com/azure-sdk/internal/_packaging/azure-sdk-for-java-pr/maven/v1
+            if ('$(Build.Repository.Name)' -eq 'Azure/azure-sdk-for-java') {
+              # Publish to the public feed
+              # Publish to https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-java/maven/v1
+              echo "##vso[task.setvariable variable=RepositoryUrl]https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-java/maven/v1"
+              echo "Using Java Public Dev Feed"
+            } else {
+              echo "Using Java Private Dev Feed"
+            }
+          displayName: Setup TargetFeed
+
+        - template: tools/gpg/gpg.yml@azure-sdk-build-tools
+
+        - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
[in follow-up] We should do this publishing as a batch.

Reply:
This one was marked twice and is covered in the bug, https://github.com/azure/azure-sdk-for-java/issues/28662, created to deal with publish in general.